### PR TITLE
Adding Releases Page to Docs

### PR DESCRIPTION
**Pull request by robscott**
This should fix https://github.com/kubernetes-sigs/service-apis/issues/10.

/assign @bowei

### CODE REVIEW

File: docs-src/releases.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+a great deal of API stability before releasing a beta.
+
+Conversions between different versions would not be straightforward, so in
+general, alpha would be used to verify that controllers can work with this API
+and that it makes sense, and a beta release would be the first version consumed
+by most end users. When upgrading to beta, we can remove alpha versions from the
+list of supported versions and require anyone with alpha resources (should be
+very few) to replace them with beta resources.
+
+Although VolumeSnapshot CRDs could be installed directly by CSI drivers that
```
Comment (youngnick) [inline docs-src/releases.md]: s/VolumeSnapshot/something/ ?

File: docs-src/releases.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+being done:
+[VolumeSnapshots](https://kubernetes.io/blog/2018/10/09/introducing-volume-snapshot-alpha-for-kubernetes/). These definitions are
+automatically installed by the CSI drivers that support them. This pattern
+doesn't make as much sense here where we want to support multiple controllers
+per cluster.
+
+CRDs do not need to be tied to Kubernetes versions, the release cycle is
+independent. The primary requirement of an official CRD is that it goes through
+the KEP process. Versioning should be well thought out, with several alphas, and
+a great deal of API stability before releasing a beta.
```
Comment (bowei) [inline docs-src/releases.md]: We probably want to be more specific here. Isn't this basically the Kubernetes API stability contract? Maybe just reference that, but with a faster iteration period than k8s core release cadence?

File: docs-src/releases.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+Conversions between different versions would not be straightforward, so in
```
Comment (bowei) [inline docs-src/releases.md]: Can we make the language more direct:

```
* Alpha will be released on a fast cadence (monthly). Users and controller authors will be expected to use the latest version of the API. There will be little to no provisions for backwards compatibility. Generally we expect the alpha API to be for users and controller developers to test out the API but not in any production environment.
* Beta will meet the Kubernetes API compatibility guarantees and inter operation. etc etc
```

File: docs-src/releases.md (hunk 1 above)
Comment (bowei) [inline docs-src/releases.md]: We should probably say they will be released from this repo?

File: docs-src/releases.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+# Releases
+
+## Official Custom Resource Definitions
+
+The custom resources defined as part of Service APIs will be considered official
+custom resource definitions (CRDs). That means that they will go through a KEP
+review process and be part of a Kubernetes project. 
+
+This is a relatively new concept, and there is only one previous example of this
```
Comment (bowei) [inline docs-src/releases.md]: Let's move this to a "discussion" section below and put a straightforward description of the release policy/process up front.

File: docs-src/releases.md (hunk 1 above)
Comment (robscott) [reply docs-src/releases.md]: Yeah that was unnecessarily confusing. I'd mostly copied this from the corresponding issue where I was talking about how we could learn from the example of VolumeSnapshot (the only other example of an "official CRD"). I was trying to contrast that with what would work here. I've reworked the docs a bunch here to actually be more of a releases page than just a recap of that conversation.

File: docs-src/releases.md (hunk 2 above)
Comment (robscott) [reply docs-src/releases.md]: Updated, thanks! What I linked to does not talk about release cadence, just API versioning requirements.

File: docs-src/releases.md (hunk 3 above)
Comment (robscott) [reply docs-src/releases.md]: Updated, thanks!

File: docs-src/releases.md (hunk 1 above)
Comment (robscott) [reply docs-src/releases.md]: Updated, thanks!

File: docs-src/releases.md (hunk 4 above)
Comment (robscott) [reply docs-src/releases.md]: With all the changes I made to the structure here, I ended up using a "other official custom resources" section at the bottom to talk about VolumeSnapshots. Hopefully that makes sense here.

File: docs-src/releases.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+APIs, these APIs will not be installed by default on Kubernetes clusters. This
+project will use custom resource definitions (CRDs) to represent the new API
+types that Service APIs include.
+
+Similar to other Kubernetes APIs, these will go through a formal Kubernetes
+Enhancement Proposal (KEP) review. Unlike other Kubernetes APIs, Service API
+releases will be independent from Kubernetes releases.
+
+Service API releases will include four components:
+- Custom resource definitions to define the API
```
Comment (danehans) [inline docs-src/releases.md]: s/Custom resource definitions/Custom Resource Definitions (CRDs).

File: docs-src/releases.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+- Go client libraries
+- Validation webhooks to implement cross field validations
+- Conversion webhooks to convert resources between API versions
```
Comment (danehans) [inline docs-src/releases.md]: The formatting of this bulleted listis not being generated as you expect. Run `make serve` to see yourself. See https://daringfireball.net/projects/markdown/basics

File: docs-src/releases.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+## Versioning
+
+Versioning will be completely separate from the Kubernetes release process, but
+similar methodology will be used. Service API versions will use the [same
+version level requirements as other Kubernetes
+features](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions).
```
Comment (danehans) [inline docs-src/releases.md]: Is service-apis considered development level version in https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions? If so, maybe this doc should call this out.

File: docs-src/releases.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+Beta and stable releases will operate on a slower, more standard, release
+schedule. They will also provide all of the stability guarantees that other beta
+and stable Kubernetes features provide.
+
+## Installation
+
+This project will be responsible for providing straightforward and reliable ways
+to install releases of Service APIs. Notably, that will need to include a
+strategy for the installation of the validation and conversion webhooks.
```
Comment (danehans) [inline docs-src/releases.md]: >  Notably, that will need to include a strategy for the installation of the validation and conversion webhooks.
I get a bit confused by the wording here. Is this more of a todo of the project? If so, maybe:

__Note:__ An installation strategy for validation and conversion webhooks will be provide in the futre.

File: docs-src/releases.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+## Other Official Custom Resources
+
+This is a relatively new concept, and there is only one previous example of
+official custom resources being used being done:
```
Comment (danehans) [inline docs-src/releases.md]: s/being used being done/being used

File: docs-src/releases.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+[VolumeSnapshots](https://kubernetes.io/blog/2018/10/09/introducing-volume-snapshot-alpha-for-kubernetes/).
+Although VolumeSnapshot CRDs can be installed directly by CSI drivers that
+support them, we need to support multiple controllers per cluster, so the CRDs
```
Comment (danehans) [inline docs-src/releases.md]: s/we need to support/Service APIs must support/

File: docs-src/releases.md [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Releases
+
+Although Service APIs are an official Kubernetes project, and represent official
+APIs, these APIs will not be installed by default on Kubernetes clusters. This
```
[the remark this answers is not in the document]
Comment (robscott) [reply docs-src/releases.md]: Updated, thanks!

File: docs-src/releases.md (hunk 5 above)
Comment (robscott) [reply docs-src/releases.md]: Updated, thanks!

File: docs-src/releases.md [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+project will use custom resource definitions (CRDs) to represent the new API
+types that Service APIs include.
+
+Similar to other Kubernetes APIs, these will go through a formal Kubernetes
+Enhancement Proposal (KEP) review. Unlike other Kubernetes APIs, Service API
+releases will be independent from Kubernetes releases.
```
[the remark this answers is not in the document]
Comment (robscott) [reply docs-src/releases.md]: Updated, thanks!

File: docs-src/releases.md (hunk 6 above)
Comment (robscott) [reply docs-src/releases.md]: :man_facepalming: woops, thanks!

File: docs-src/releases.md (hunk 7 above)
Comment (robscott) [reply docs-src/releases.md]: Good call, added a line about that.

File: docs-src/releases.md (hunk 8 above)
Comment (robscott) [reply docs-src/releases.md]: Yeah that was confusing, just removed that for now. If needed we can open an issue to discuss the different options for webhook installation.

File: docs-src/releases.md (hunk 9 above)
Comment (robscott) [reply docs-src/releases.md]: woops, thanks!

File: docs-src/releases.md (hunk 10 above)
Comment (robscott) [reply docs-src/releases.md]: Updated, thanks!

**Pull request merged**